src/internal/pjs/api_server.go

```diff
@@ -189,3 +176,108 @@ func (a *apiServer) ProcessQueue(srv pjs.API_ProcessQueueServer) (retErr error)
 		}
 	}
 }
+
+func (a *apiServer) WalkJob(req *pjs.WalkJobRequest, srv pjs.API_WalkJobServer) (err error) {
+	ctx := pctx.Child(srv.Context(), "walkJob")
```

Comment:
FWIW, srv.Context() is already named WalkJob (by grpc interceptors, when called thru grpc).

I like log.SpanContext for per-operation contexts.  You get a child context that you get to name, but it automatically reports success/failure, how long the deadline on the context is (or its error if it's done), etc.